plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/core.py

```diff
@@ -0,0 +1,228 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+import json
+import logging
+from typing import Any, Dict, List, Optional, Union
+
+import nevergrad as ng  # type: ignore
+from omegaconf import DictConfig, OmegaConf
+
+from hydra.core.config_loader import ConfigLoader
+from hydra.core.config_search_path import ConfigSearchPath
+from hydra.core.plugins import Plugins
+from hydra.plugins.launcher import Launcher
+from hydra.plugins.search_path_plugin import SearchPathPlugin
+from hydra.plugins.sweeper import Sweeper
+from hydra.types import TaskFunction
+
+log = logging.getLogger(__name__)
+
+
+class NevergradSearchPathPlugin(SearchPathPlugin):
+    """
+    This plugin is allowing configuration files provided by this sweeper plugin to be discovered
+    and used once it is installed.
+    """
+
+    def manipulate_search_path(self, search_path: ConfigSearchPath) -> None:
+        # Appends the search path for this plugin to the end of the search path
+        search_path.append(
+            "nevergrad-sweeper", "pkg://hydra_plugins.hydra_nevergrad_sweeper.conf"
+        )
+
+
+def convert_to_deduced_type(string: str) -> Union[int, float, str]:
+    """Converts a string into a float or an int if it makes sense,
+    or returns the string.
+
+    Examples
+    --------
+    read_value("1")
+    >>> 1
+
+    read_value("1.0")
+    >>> 1.0
+
+    read_value("blublu")
+    >>> "blublu"
+    """
+    output: Union[float, int, str] = string
+    try:
+        output = float(string)
+    except ValueError:
+        pass
+    else:
+        if output.is_integer() and "." not in string:
+            output = int(output)
+    return output
```

Comment:
I am not familiar with try, except, else syntax.
Does else execute even though you you pass the except?

Do you have any tests covering this? in particular I have doubts about the int case.

Reply:
else execute only if their is no Exception, there is a `finally` clause if one want something that always executes in the end.
This is tested (indirectly) through `test_make_parameter` (see `value_cls`), I can crosscheck tomorrow though.